libs/ui/select/brain/src/lib/brn-select.service.ts

```diff
@@ -92,4 +95,51 @@ export class BrnSelectService {
 	public _setSelectTrigger(trigger: BrnSelectTriggerDirective) {
 		this._selectTrigger = trigger;
 	}
+
+	public registerOption(option: CdkOption | null) {
+		this.possibleOptions.update((options) => [...options, option]);
+	}
+
+	public deregisterOption(option: CdkOption | null) {
+		this.possibleOptions.update((options) => options.filter((o) => o !== option));
+	}
+
+	public selectOptionByValue(value: unknown) {
+		if (value === null || value === undefined) {
+			this.state.update((state) => ({
+				...state,
+				selectedOptions: [],
+				value: this.multiple() ? [] : '',
+			}));
+			return;
+		}
+
+		const options = this.possibleOptions();
+
+		if (typeof value === 'string' || Array.isArray(value)) {
```

Comment:
@badsgahhl do you know why you added this ```if``` ? 
This seems to break the initial value of non strings, when bound to ngModel.